metadeploy/api/views.py

```diff
@@ -207,20 +220,80 @@ def preflight(self, request, pk=None):
+        valid_plan_type = (
+            plan.supported_orgs == SUPPORTED_ORG_TYPES.Scratch
+            or plan.supported_orgs == SUPPORTED_ORG_TYPES.Both
+        )
+        if not valid_plan_type:
+            return Response(
+                {"detail": "This plan does not support creating a scratch org."},
+                status=status.HTTP_409_CONFLICT,
+            )
+
+        new_data = request.data.copy()
+        new_data["plan"] = str(plan.pk)
+        serializer = ScratchOrgJobSerializer(data=new_data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        # Check queue status
+        # If overfull, return 503
+        queue = django_rq.get_queue("default")
+        if len(queue) > settings.MAX_QUEUE_LENGTH:
+            return Response(
+                {"detail": "Queue is overfull. Try again later."},
+                status=status.HTTP_503_SERVICE_UNAVAILABLE,
+            )
+
+        uuid = str(uuid4())
+        scratch_org = serializer.save(uuid=uuid)
+        request.session["scratch_org_id"] = uuid
+
+        serializer = ScratchOrgJobSerializer(instance=scratch_org)
+        return Response(serializer.data, status=status.HTTP_202_ACCEPTED,)
+
+    @action(detail=True, methods=["post", "get"], permission_classes=(AllowAny,))
+    def scratch_org(self, request, pk=None):
+        if request.method == "GET":
+            pass
+            # @@@ return self.scratch_org_get(request)
+        if request.method == "POST":
+            return self.scratch_org_post(request)
+
 
 class OrgViewSet(viewsets.ViewSet):
+    permission_classes = (AllowAny,)
+
     def list(self, request):
         """
         This will return data on the user's current org. It is not a
         list endpoint, but does not take a pk, so we have to implement
         it this way.
         """
-        current_job = Job.objects.filter(
-            org_id=request.user.org_id, status=Job.Status.started
-        ).first()
-        current_preflight = PreflightResult.objects.filter(
-            org_id=request.user.org_id, status=PreflightResult.Status.started
-        ).first()
+        if "scratch_org_id" in request.session:
+            uuid = request.session["scratch_org_id"]
+            current_job = Job.objects.filter(
+                uuid=uuid, status=Job.Status.started
+            ).first()
+            current_preflight = PreflightResult.objects.filter(
+                uuid=uuid, status=PreflightResult.Status.started
+            ).first()
+        elif request.user.is_authenticated:
+            current_job = Job.objects.filter(
+                org_id=request.user.org_id, status=Job.Status.started
+            ).first()
+            current_preflight = PreflightResult.objects.filter(
+                org_id=request.user.org_id, status=PreflightResult.Status.started
+            ).first()
+        else:
+            return Response(status=status.HTTP_403_FORBIDDEN)
```

Comment:
@wlonk Since the front-end can't know whether `request.session` has a valid UUID, I think we want this to return empty (e.g. `{"current_job": None, "current_preflight": None}` rather than a 403.